docs/_docs/modules.md

```diff
@@ -0,0 +1,90 @@
+| --------------------- | ----------------------------- |
+| @Design               | @DesignProperty               |
+| @DesignVariant        | @DesignVariantProperty        |
+| @DesignContentTypes   | @DesignContentTypesProperty   |
+| @DesignPreviewContent | @DesignPreviewContentProperty |
+
+Note that once this bug has been fixed, all of the `@Design*Property` annotations will be deprecated for at least one release, and then later removed, in favor of the analogous annotations used for functions.
+
+### Example:
+```kotlin
+@DesignModuleClass
+class TextModule(
+    @DesignProperty(node = "#text") val text: String,
+    @DesignProperty(node = "#replace")
+    val replaceNode: @Composable (ComponentReplacementContext) -> Unit,
+    @DesignVariantProperty(property = "#text-style") val textStyle: TextStyle,
+    @DesignContentTypesProperty(nodes = ["#button", "#header"])
+    @DesignPreviewContentProperty(
+        name = "Buttons",
+        nodes = [PreviewNode(3, "#button")]
+    )
+    val content: ListContent,
+)
+```
+
+## Nested Modules
+Module classes can include other modules. The resulting set of customizations will contain all customizations from the current class as well as all nested module classes. Use the `@DesignModuleProperty` annotation to achieve this, using the class name directly as the type of the `@DesignModuleProperty` class property.
+
+### Example:
+```kotlin
+@DesignModuleClass
+class TextModuleCombined(
+    @DesignProperty(node = "#name") val name: String,
+    @DesignModuleProperty val textProperties: TextModule,
+)
+```
+
+## Usage in a function
+Module classes by themselves are not useful until used as a parameter in a `@DesignComponent` function. Use the `@DesignModule` annotation for parameters in such a function. The generated function will combine any customizations from `@Design` and `@DesignVariant` parameters in the function with all the embedded customizations from parameters that are modules.
+
+### Example:
+```kotlin
+interface ModuleExample {
+    @DesignComponent(node = "#stage")
+    fun Main(
+        @Design(node = "#header") header: String,
+        @DesignModule moduleCustomizations: TextModuleCombined,
+    )
+}
+```
+
+To call the generated function, an instance of TextModuleCombined must be passed into the `moduleCustomizations` parameter. This can be constructed normally since it uses the developer defined class directly, not a generated one.
+
+```kotlin
+fun getModuleCustomizations(): TextModuleCombined {
+    return TextModuleCombined(
+        name = "Combined Module",
+        textProperties = TextModule(
+            text = "Hello World",
+            replaceNode = { ModuleExample.Button("My Button") },
+            textStyle = TextStyle.LargeBold,
+            content = {
+                ListContentData(count = 3) { index ->
+                    ModuleExample.Button("Button $index")
+                }
+            },
+        ),
+    )
+}
```

Comment:
Because that's how I would imagine it would often work when using modules, since a module would likely be shared and you'd have some common code to construct one. I don't really feel strongly about this either way though, so I've removed it.